### PR TITLE
293: Save user id on translation status change

### PR DESCRIPTION
**Pull request by yoavf**
This adds value for last modified user id to all status change of translations.
See #293

### CODE REVIEW

**Comment by yoavf** [comment]
(In progress, not ready for review yet)

**Comment by yoavf** [comment]
screen shot 2016-06-06 at 11 11 53

^ out of date now, not showing both if by the same user

**Comment by yoavf** [comment]
Do we want an index, or let's wait for an application before adding it?

Re travis-ci: do we really want do add docs do all the self-explanatory object properties? I can do that, but feels a bit useless.

**Comment by toolstack** [comment]
I don't see any reason to add an index at this point, if we add a feature that needs it we can add the index at that point.

As for the docs, yes document everything.

**Comment by ocean90** [comment]
> Do we want an index, or let's wait for an application before adding it?

I don't think it's necessary at the moment.

> Re travis-ci: do we really want do add docs do all the self-explanatory object properties? I can do that, but feels a bit useless.

Docs are never useless. :)

File: gp-includes/things/translation.php [hunk 1]
```diff
@@ -31,6 +31,14 @@ class GP_Translation extends GP_Thing {
 	public $translation_4;
 	public $translation_5;
 	public $user_id;
+	/**
+	 * User id of user (validator) who last changed the status of the translation.
+	 *
+	 * @since 2.1.0
+	 *
+	 * @var $user_id_last_modified
```
Comment (ocean90) [inline gp-includes/things/translation.php]: Should be `@var int`, see https://phpdoc.org/docs/latest/references/phpdoc/tags/var.html.

File: gp-includes/template-links.php [hunk 2]
```diff
@@ -193,3 +193,25 @@ function gp_link_glossary_delete_get( $glossary, $set, $text = false, $attrs = a
 function gp_link_glossary_delete( $glossary, $set, $text = false, $attrs = array() ) {
 	echo gp_link_glossary_delete_get( $glossary, $set, $text, $attrs );
 }
+
+/**
+ * Outputs a HTML link to a user profile page.
+ *
+ * @since 2.1.0
+ *
+ * @param WP_USER $user user object.
```
Comment (ocean90) [inline gp-includes/template-links.php]: `@param WP_User $user A WP_User object.`

File: gp-templates/translation-row.php [hunk 3]
```diff
@@ -145,26 +145,26 @@
...
+				<dl>
+					<dt><?php
+						if ( 'current' === $t->translation_status ) {
+							_e( 'Approved by:', 'glotpress' );
+						} else if ( 'rejected' === $t->translation_status ) {
+							_e( 'Rejected by:', 'glotpress' );
+						} else {
+							_e( 'Last updated by:', 'glotpress' );
+						}
+						?>
```
Comment (ocean90) [inline gp-templates/translation-row.php:164]: There is a closing `</dt>` tag missing.

**Comment by ocean90** [comment]
@yoavf Are you still working on this or can someone else take over?

**Comment by yoavf** [comment]
> @yoavf Are you still working on this or can someone else take over?

I would, but in the middle of moving houses right now. I can resume on
Sunday, but I don't mind if someone else takes over before that. Thanks ☺️

**Comment by ocean90** [comment]
New PR: #481

**Pull request closed without merging**